vmm/src/device_manager.rs

```diff
@@ -430,6 +430,12 @@ pub struct DeviceManager {
     // The virtio devices on the system
     virtio_devices: Vec<(VirtioDeviceArc, bool)>,
 
+    // List of bus devices
+    // Let the DeviceManager keep strong references to the BusDevice devices.
+    // This allows the IO and MMIO buses to be provided with Weak references,
+    // which prevents from cyclic dependencies.
```

Comment:
nit: s/from//

Reply:
At some point we should do an audit and find which devices are only on this vector. Long term for hot-unplug all the PCI devices will be stored somewhere else as is the serial/ioapic/ged device.